ocaml-lsp-server/src/ocamlformat.ml

```diff
@@ -0,0 +1,118 @@
+open Import
+
+module File_type = struct
+  type t =
+    | Impl
+    | Intf
+    | Name of string
+
+  let to_cmdline_args = function
+    | Impl -> [ "--impl" ]
+    | Intf -> [ "--intf" ]
+    | Name n -> [ Printf.sprintf "--name=%s" n ]
+end
+
+module Input = struct
+  type t =
+    | Stdin of string * File_type.t
+    | File of string
+end
+
+module Output = struct
+  type _ t =
+    | File : string -> unit t
+    | Stdout : string t
+
+  let to_cmdline_args : type o. o t -> string list = function
+    | File s -> [ Printf.sprintf "--output=%s" s ]
+    | Stdout -> []
+end
+
+module Options = struct
+  type t = string list
+
+  let default : t = []
+end
+
+type 'result command =
+  | Format_file : Input.t * 'result Output.t -> 'result command
+  | Format_files_in_place : string list -> unit command
+  | Check : Input.t -> bool command
+
+let read_to_end ?(hint = 0) (in_chan : in_channel) : string =
+  let buf = Buffer.create hint in
+  let rec go () =
+    let line = input_line in_chan in
+    Buffer.add_string buf line;
+    Buffer.add_char buf '\n';
```

Comment:
This seems to change the user's line endings to `\n` even if they're using `\r\n`. I think that is quite annoying for Windows users. How about using `input` instead of `input_line`?